test/minitest/test_minitest_benchmark.rb

```diff
@@ -118,6 +118,54 @@ def test_fit_power_noisy
     assert_fit :power, x, y, 0.96, 3.119e-5, 0.8959
   end
 
+  def test_sigma
+    bench = Minitest::Benchmark.new :blah
+    assert_equal bench.sigma([1,2,3]), 6
+    assert_equal bench.sigma([1,2,3]) { |n| n**2 }, 14
+  end
+
+  def test_fit_error
+    bench = Minitest::Benchmark.new :blah
+    assert_equal bench.fit_error([[1, 1], [2, 2], [3, 3]]) { |x| x }, 1.0
+    assert_in_delta bench.fit_error([[1, 1], [2, 2], [3, 4]]) { |x| x }, 0.785
+  end
+
+  def test_fit_linear_x2
+    bench = Minitest::Benchmark.new :blah
+    xs = [1, 10, 100, 1000]
+    _a, _b, rr = bench.fit_linear(xs, xs.map { |x| x**2 }), :<, 0.99
```

Comment:
I dunno what this line is supposed to be doing, but if you remove "`, :<, 0.99`" it'll at least provide the real `rr`. As it currently stands, `_a` is populated with the 3 values from `fit_linear`, `_b` gets `:<`, and `rr` gets `0.99`, which means the test always fails.

Reply:
Both of these `fit_linear` tests (x2 and x3) pass if given a linear range... You're giving it an exponential range but expecting a linear response in `y`. In this test, if you get the real `rr` back, for a range of `1..10`, the `rr` is `0.94` which is what wolfram alpha says it should be:

http://www.wolframalpha.com/input/?i=linear+fit+1,+4,+9,+16,+25,+36,+49,+64,+81,+100

(that's so cool!)

Where with your exponential `xs`, mathematica (I couldn't figure out how to specify custom x values) agrees with minitest's argorithm that it fits:

```
In[28]:= LinearModelFit[{{1, 1}, {10, 100}, {100, 10000}, {1000, 
    1000000}}, x, x]["RSquared"]

Out[28]= 0.993096

In[29]:= LinearModelFit[{{1, 1}, {2, 100}, {3, 10000}, {4, 1000000}}, 
  x, x]["RSquared"]

Out[29]= 0.607999
```